### PR TITLE
Feature/404page

### PR DESCRIPTION
**Pull request by Hunter27**
Created error page and implemented loading icon

### CODE REVIEW

File: waterlog/src/Components/Map.js [hunk 1]
```diff
@@ -1,42 +1,44 @@
...
     todaysLeaks = [];
   } else {
-    todaysLeaks = data.leaks.filter(leak => (
-      (new Date(leak.latestTimeStamp)).getDay() === (new Date(Date.now())).getDay() &&
-      (new Date(leak.latestTimeStamp)).getMonth() === (new Date(Date.now())).getMonth() &&
-      (new Date(leak.latestTimeStamp)).getFullYear() === (new Date(Date.now())).getFullYear()
-    ));
+    todaysLeaks = data.leaks.filter(
+      leak =>
+        new Date(leak.latestTimeStamp).getDay() ===
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js]: those are a lot of repeated calls look at putting `Date(Date.now())` into a variable

File: waterlog/src/Components/Map.js [hunk 2]
```diff
@@ -47,75 +49,88 @@ function formatMapData(data) {
...
+    if (
+      Object.keys(sensorIn) === validSensor ||
+      Object.keys(sensorOut) === validSensor
+    )
+      return <div />;
     let sensorInColor = defaultColor,
       sensorOutColor = defaultColor,
       segmentColor = defaultColor;
-    if (sensorIn.status.toLowerCase() === 'fault') {
+    if (sensorIn.status.toLowerCase() === "fault") {
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js:79]: Be consistent, either `if (bool) { oneLiner }` or `if () oneLiner`

File: waterlog/src/Components/Wastage.js [hunk 3]
```diff
@@ -1,105 +1,126 @@
...
+  };
 
-	getGraphType = () => {
-		if (this.state.display === 'daily') return <DailyWastageComponent props={this.props.dailyWaste} />;
-		else if (this.state.display === 'monthly') return <MonthlyWastageComponent props={this.props.monthlyWaste} />;
-		else if (this.state.display === 'seasonal') return <SeasonalWastageComponent props={this.props.seasonWaste} />;
-		else return <div>Error has occured</div>;
-	};
+  getGraphType = () => {
+    if (this.state.display === "daily")
```
Comment (FredJEhlers) [inline waterlog/src/Components/Wastage.js]: same comment about if statements from before

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/Map.js (hunk 2 above)
Comment (Hunter27) [reply waterlog/src/Components/Map.js:79]: done

File: waterlog/src/Components/Wastage.js (hunk 3 above)
Comment (Hunter27) [reply waterlog/src/Components/Wastage.js]: done

File: waterlog/src/Components/Map.js (hunk 1 above)
Comment (Hunter27) [reply waterlog/src/Components/Map.js]: done

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**